From 0617db0a21146d91760f978c97ed318d4b43f2dd Mon Sep 17 00:00:00 2001
From: Hall Liu <hallliu@google.com>
Date: Tue, 12 Jun 2018 11:29:32 -0700
Subject: [PATCH 02/70] Mark RTT calls as non-conferenceable

In refreshConferenceSupported, check if the call is RTT and mark it
non-conferenceable if so. Also call refreshConferenceSupported when the
RTT status changes.

Bug: 109763648
Test: manual
Change-Id: I374b737d4f5dabec3091fb2da2eaf8355d54d221
---
 .../services/telephony/TelephonyConnection.java       | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/com/android/services/telephony/TelephonyConnection.java b/src/com/android/services/telephony/TelephonyConnection.java
index d6fb3ad57..2b0f49fb5 100644
--- a/src/com/android/services/telephony/TelephonyConnection.java
+++ b/src/com/android/services/telephony/TelephonyConnection.java
@@ -538,6 +538,7 @@ abstract class TelephonyConnection extends Connection implements Holdable {
         @Override
         public void onRttModifyResponseReceived(int status) {
             updateConnectionProperties();
+            refreshConferenceSupported();
             if (status == RttModifyStatus.SESSION_MODIFY_REQUEST_SUCCESS) {
                 sendRttInitiationSuccess();
             } else {
@@ -554,7 +555,12 @@ abstract class TelephonyConnection extends Connection implements Holdable {
 
         @Override
         public void onRttInitiated() {
-            updateConnectionProperties();
+            if (mOriginalConnection != null) {
+                // if mOriginalConnection is null, the properties will get set when
+                // mOriginalConnection gets set.
+                updateConnectionProperties();
+                refreshConferenceSupported();
+            }
             sendRttInitiationSuccess();
         }
 
@@ -2117,6 +2123,9 @@ abstract class TelephonyConnection extends Connection implements Holdable {
         if (mTreatAsEmergencyCall) {
             isConferenceSupported = false;
             Log.d(this, "refreshConferenceSupported = false; emergency call");
+        } else if (isRtt()) {
+            isConferenceSupported = false;
+            Log.d(this, "refreshConferenceSupported = false; rtt call");
         } else if (!isConferencingSupported || isIms && !isImsConferencingSupported) {
             isConferenceSupported = false;
             Log.d(this, "refreshConferenceSupported = false; carrier doesn't support conf.");
-- 
2.17.1

